lib/game_modules/matrix_board.ex

```diff
@@ -0,0 +1,64 @@
+defmodule MatrixBoard do
+  def create_board(row_size) do
+    nil
+    |> List.duplicate(row_size)
+    |> List.duplicate(row_size)
+  end
+
+  def columns(board) do
+    board
+    |> Enum.zip()
+    |> Enum.map(&Tuple.to_list/1)
+  end
+
+  def rows(board) do
+    board
+  end
+
+  def diagonals(board) do
+    [
+      get_diagonal(board),
+      get_diagonal(Enum.reverse(board))
+    ]
+  end
+
+  def get_row_at(board, row) do
+    Enum.at(rows(board), row)
+  end
+
+  def get_col_at(board, col) do
+    Enum.at(columns(board), col)
+  end
+
+  def has_winning_row?(board) do
+    check_for_win(rows(board))
+  end
+
+  def has_winning_column?(board) do
+    check_for_win(columns(board))
+  end
+
+  def has_winning_diagonal?(board) do
+    check_for_win(diagonals(board))
+  end
+
+  def is_winning_row?(row) do
+    Enum.count(Enum.uniq(row)) == 1
+  end
+
+  def is_full?(board) do
+    board
+    |> List.flatten
+    |> Enum.all?(fn x -> x in [:x, :o] end)
```

Comment:
The hardcoded `:x` and `:o` here are a little bit like the [magic numbers](https://stackoverflow.com/questions/47882/what-is-a-magic-number-and-why-is-it-bad) code smell. This module works beautifully for boards of any size, containing any kind of marker. Why limit this method to `:x` and `:o` markers by hardcoding them here? Can you think of a way to check if the board is full that doesn't rely on the `:x` and `:o` atoms? 

In general, it's a good idea to treat your abstractions like they work in the CIA: they should only have as much information as they need to do their job and operate on a need-to-know basis. Does `Board` need to know about the `:x` and `:o` atoms to function? If not, they probably belong somewhere else.